src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp

```diff
@@ -113,7 +122,85 @@ class boss_broodlord : public CreatureScript
     }
 };
 
+class go_suppression_device : public GameObjectScript
+{
+    public:
+        go_suppression_device() : GameObjectScript("go_suppression_device") { }
+
+        struct go_suppression_deviceAI : public GameObjectAI
+        {
+            go_suppression_deviceAI(GameObject* go) : GameObjectAI(go), _instance(go->GetInstanceScript()) { }
+
+            void InitializeAI() override
+            {
+                _events.ScheduleEvent(EVENT_SUPPRESSION_AURA, Seconds(6));
+                _events.ScheduleEvent(EVENT_SUPPRESSION_ANIM, Seconds(2));
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (_instance->GetBossState(DATA_VAELASTRAZ_THE_CORRUPT) != DONE)
+                    return;
+
+                if (me->GetGoState() == GO_STATE_DESTROYED)
```

Comment:
This will override state back to GO_READY on next UpdateAI even if DATA_BROODLORD_LASHLAYER is DONE, you would need to move this part after that check

Reply:
Thank you! :)